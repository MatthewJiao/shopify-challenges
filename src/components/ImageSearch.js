import React, { useState } from 'react';

const ImageSearch = ({ searchText }) => {
  const [text, setText] = useState('');

  const onSubmit = (e) => {
    e.preventDefault();
    console.log('test')
    searchText(text);
  }

  return (
    <div style = {searchStyle} className='max-w-sm rounded overflow-hidden my-10 mx-auto'>
      <form style = {searchContainerStyle} onSubmit={onSubmit} className="w-full max-w-sm">
        <div className="flex items-center border-b border-b-2 border-teal-500 py-2">
          <input style = {inputStyle} onChange={e => setText(e.target.value)} className="appearance-none bg-transparent border-none w-full text-gray-900 mr-3 py-1 px-2 leading-tight focus:outline-none" type="text" placeholder="Search ..." />
          <button style = {buttonStyle} className="rounded-md text-sm border-2 py-1 px-2" type="submit">
            Search
          </button>
        </div>
      </form>
		</div>
  )
}

const searchContainerStyle = {

}

const searchStyle = {
  marginTop: "5rem"
}

const buttonStyle = {
  color: "white",
  backgroundColor: "#6DCBC5",
  borderColor: "white",
  fontSize: "1rem",
  padding: "0.5rem"
}

const inputStyle = {
  color: "#6DCBC5",
  fontSize: "1.6rem",
  placeholder: { 
    color: "red",
    opacity: "1"
  }
  
}

export default ImageSearch;