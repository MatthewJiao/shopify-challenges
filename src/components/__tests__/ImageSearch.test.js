import React from 'react'
import reactDom from 'react-dom'
import { cleanup, render } from '@testing-library/react'
import renderer from 'react-test-renderer'

import ImageSearch from '../ImageSearch'


afterEach(cleanup)
it("image search renders properly and without issue", () => {
    const div = document.createElement("div");
    reactDom.render(<ImageSearch searchText = {"kittens"} />, div);
})


it("matches snapshot", () => {
    const tree = renderer.create(<ImageSearch searchText = {"kittens"} />).toJSON();
    expect(tree).toMatchSnapshot()
})
