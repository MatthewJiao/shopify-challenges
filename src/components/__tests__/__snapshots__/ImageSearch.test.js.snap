// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="max-w-sm rounded overflow-hidden my-10 mx-auto"
  style={
    Object {
      "marginTop": "5rem",
    }
  }
>
  <form
    className="w-full max-w-sm"
    onSubmit={[Function]}
    style={Object {}}
  >
    <div
      className="flex items-center border-b border-b-2 border-teal-500 py-2"
    >
      <input
        className="appearance-none bg-transparent border-none w-full text-gray-900 mr-3 py-1 px-2 leading-tight focus:outline-none"
        onChange={[Function]}
        placeholder="Search ..."
        style={
          Object {
            "color": "#6DCBC5",
            "fontSize": "1.6rem",
            "placeholder": Object {
              "color": "red",
              "opacity": "1",
            },
          }
        }
        type="text"
      />
      <button
        className="rounded-md text-sm border-2 py-1 px-2"
        style={
          Object {
            "backgroundColor": "#6DCBC5",
            "borderColor": "white",
            "color": "white",
            "fontSize": "1rem",
            "padding": "0.5rem",
          }
        }
        type="submit"
      >
        Search
      </button>
    </div>
  </form>
</div>
`;
